superpoint/generate_coco_patches.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import tensorflow as tf
+import cv2 as cv
+import os
+import argparse
+import yaml
+from pathlib import Path
+
+from superpoint.models.utils import (sample_homography, flat2mat,
+                                     invert_homography)
+from superpoint.settings import DATA_PATH
+
+
+seed = None
+
+
+def _scale_preserving_resize(image):
+    target_size = tf.convert_to_tensor(config['preprocessing']['resize'])
+    scales = tf.to_float(tf.divide(target_size, tf.shape(image)[:2]))
+    new_size = tf.to_float(tf.shape(image)[:2]) * tf.reduce_max(scales)
+    image = tf.image.resize_images(image, tf.to_int32(new_size),
+                                   method=tf.image.ResizeMethod.BILINEAR)
+    return tf.image.resize_image_with_crop_or_pad(image, target_size[0],
+                                                  target_size[1])
+
+
+def _preprocess(image):
+    image = tf.image.rgb_to_grayscale(image)
+    if config['preprocessing']['resize']:
+        image = _scale_preserving_resize(image)
+    return image
+
+
+if __name__ == '__main__':
+    tf.set_random_seed(seed)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('config', type=str, default=None)
+    args = parser.parse_args()
+    with open(args.config, 'r') as f:
+        config = yaml.load(f)
+
+    base_path = Path(DATA_PATH, 'COCO/val2014/')
+    image_paths = list(base_path.iterdir())
+    output_dir = Path(DATA_PATH, 'COCO/patches/')
+    if not output_dir.exists():
+        os.makedirs(output_dir)
+
+    print("Generating patches of Coco val...")
+    sess = tf.InteractiveSession()
+    for num, path in enumerate(image_paths):
+        new_path = Path(output_dir, str(num))
+        if not new_path.exists():
+            os.makedirs(new_path)
+
+        # Read the image
+        image = tf.read_file(str(path))
```

Comment:
Are you really creating new ops at every loop iteration? Shouldn't they be instantiated before hand? I don't think that even `tf.InteractiveSession` allows this.